### ISSUE TITLE
Many benchmark results in the readme are incorrect

### ISSUE THREAD

**Issue by mohit61** [selfrefactor/rambda#608]
Hi, 

I did the benchmarking using [benchmark.js](https://benchmarkjs.com/)(which is a reliable benchmarking library) and the results were not same as mentioned in the readme, for eg. RamBda's `compose` according to benchmark result in the readme is 93.76% faster, but when I did a benchmark using benchmark.js the results of Ramda and RamBda were not much different

here is the snippet of the test suite:
```
const Benchmark = require("benchmark");
const Ra = require("ramda");
const Rb = require("rambda");


const classyGreeting = (firstName, lastName) =>
  "The name's " + lastName + ", " + firstName + " " + lastName;

const raYellGreeting = Ra.compose(Ra.toUpper, Ra.toLower, Ra.toUpper, classyGreeting);
const rbYellGreeting = Rb.compose(Ra.toUpper, Ra.toLower, Ra.toUpper, classyGreeting);

var suite = new Benchmark.Suite();

suite
  .add("Ramda compose", function () {
    raYellGreeting("James", "Bond");
  })
  .add("RamBda compose", function () {
    rbYellGreeting("James", "Bond");
  })
  .on("cycle", function (event) {
    console.log(String(event.target));
  })
  .on("complete", function () {
    console.log("Fastest is ", this.filter("fastest").map("name"));
  })
  // run async
  .run({ async: true });
```
output:
```
Ramda compose x 702,214 ops/sec ±2.08% (85 runs sampled)
RamBda compose x 739,388 ops/sec ±1.33% (88 runs sampled)
Fastest is  [ 'RamBda compose' ]
```

**Comment by selfrefactor** [comment]
I also use the same library, but I will check why results are different.

**Comment by selfrefactor** [comment]
After I checked your example versus the actual example, I see that the methods are mostly equal and my use case skews the result. 

This is known issue that I work on, namely that benchmarks require more use cases and approximation in order to get improved results.

You can check similar issue - https://github.com/selfrefactor/rambda/issues/475 and also the MR that sets the stage for the improvement - https://github.com/selfrefactor/rambda/pull/480

It is ongoing task, as it require big change in benchmark test inputs. I will keep the issue open, until I have updated `R.compose` to use the new benchmark pattern.

**Comment by selfrefactor** [comment]
I added more complex benchmarks for R.compose. I have included your example. I can confirm that your results match the outcome of the new benchmarks. I will keep the issue open until the new version is released.

**Comment by selfrefactor** [comment]
I am closing the issue as the fix is released with version `7.0.0`.
